### PR TITLE
Show error when parameter is missing in wp option patch

### LINKED ISSUE #353: Missing argument with `option patch update` should throw error
Hello, I'm having problems using this commands to disable and enable the maintenance mode of a subsite.
If the maintenance mode is enabled and i send the command to disable it the result is:
wp --url=subsite.multisite.com --allow-root option patch update seedprod_settings '{"enable_maintenance_mode":"false"}' --format=json
Success: Updated 'seedprod_settings' option.

If i send the command to enable the maintenance mode again the result is:
wp --url=subsite.multisite.com --allow-root option patch update seedprod_settings '{"enable_maintenance_mode":"true"}' --format=json
PHP Warning:  json_decode() expects parameter 1 to be string, array given in /home/multisite.com/public_html/wp-content/plugins/coming-soon/app/render-csp-mm.php on line 36
array (
  'enable_maintenance_mode' => 'true',
)

And the seedprod_settings object settings dissapear from the subsite options. Any idea what can be happening here?
Thanks in advance for the support.

### PR DESCRIPTION
**Pull request by BhargavBhandari90**
Fixes #353 

- Currently, I've displayed an error message when an argument is missing.

### CODE REVIEW

File: src/Option_Command.php [hunk 1]
```diff
@@ -569,7 +569,13 @@ function( $key ) {
 				: null;
 			$patch_value = ! empty( $stdin_value )
 				? WP_CLI::read_value( $stdin_value, $assoc_args )
-				: WP_CLI::read_value( array_pop( $key_path ), $assoc_args );
+				: ( count( $key_path ) > 1
+					? WP_CLI::read_value( array_pop( $key_path ), $assoc_args )
+					: null );
```
Comment (danielbachhuber) [inline src/Option_Command.php:574]: This nested ternary is a bit complex to read 😕

Can we expand it to some if/else statements?

File: features/option-pluck-patch.feature [hunk 2]
```diff
@@ -299,3 +299,21 @@ Feature: Option commands have pluck and patch.
...
+      }
+      """
+    And I run `wp option update option_name --format=json < input.json`
+
+    When I try `wp option patch update option_name foo`
+    And STDERR should contain:
+      """
+      Please provide value to update.
+      """
+    And the return code should be 1
```
Comment (danielbachhuber) [inline features/option-pluck-patch.feature:319]: Can we add a test case for:

```
When I run `wp option patch update option_name foo 0`
```

To make sure we're handling falsy values correctly?

Comment (BhargavBhandari90) [reply features/option-pluck-patch.feature:319]: It should not allow this to update, right?

Comment (danielbachhuber) [reply features/option-pluck-patch.feature:319]: > It should not allow this to update, right?

Perhaps I'm misunderstanding but isn't `0` a valid value?

Comment (BhargavBhandari90) [reply features/option-pluck-patch.feature:319]: Actually, I misunderstood.

I thought zero should be considered an empty value.

Then I saw other commands in the same `option-pluck-patch.feature` file.

So fixe it.

**Review by danielbachhuber — APPROVED** [review]
Thanks for the updates!

**Pull request merged**